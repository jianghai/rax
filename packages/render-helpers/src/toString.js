/**
 * Stringify an value.
 * @param val {String|Object|Function|Array} value.
 * @return {String}
 */
export default function toString(val) {
  if (val === undefined || val === null) {
    return '';
  } else {
    return typeof val === 'object'
      ? JSON.stringify(val, null, 2)
      : '' + val;
  }
};
