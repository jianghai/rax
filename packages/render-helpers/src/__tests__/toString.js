import toString from '../toString';

describe('toString', () => {
  describe('types', () => {
    it('should accept number', () => {
      expect(toString(1)).toEqual('1');
    });

    it('should accept string', () => {
      expect(toString('hello')).toEqual('hello');
    });

    it('should accept undefined', () => {
      expect(toString(undefined)).toEqual('');
    });

    it('should accept null', () => {
      expect(toString(null)).toEqual('');
    });

    it('should accept object', () => {
      expect(toString({ a: 1 })).toEqual(JSON.stringify({ a: 1 }, null, 2));
    });

    it('should accept array', () => {
      expect(toString([0, 1, 2])).toEqual(JSON.stringify([0, 1, 2], null, 2));
    });

    it('should accept function', () => {
      expect(toString(function() {
        return 'hello world';
      }))
        .toMatchSnapshot();
    });
  });
});
