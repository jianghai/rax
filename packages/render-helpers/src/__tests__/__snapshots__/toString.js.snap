// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toString types should accept function 1`] = `
"function () {
        return 'hello world';
      }"
`;
